tests/jenkins/TestVars.groovy

```diff
@@ -22,6 +22,12 @@ class TestVars extends BuildPipelineTest {
         binding.setVariable('ARTIFACT_BUCKET_NAME', 'artifact-bucket')
         binding.setVariable('AWS_ACCOUNT_PUBLIC', 'account')
         binding.setVariable('STAGE_NAME', 'stage')
+
+        helper.registerAllowedMethod("withCredentials", [List, Closure], { list, closure ->
```

Comment:
The default implementation in the unit test library substitutes the variable name `BUILD_NOTICE_WEBHOOK` by itself, this makes it fallback to globally defined variables.